### DISCUSSION TITLE
[Help] - How to force a refresh of the content

### DISCUSSION THREAD

**Discussion by boeledi** [doonfrs/pluto_grid_plus#10] [Q&A]
Hi,

First of all, many thanks for maintaining this package.

I have been looking everywhere for a solution to force a refresh of the content of the grid but could not find any "official one".

Case study:
* I have a form to create a new entry or edit an existing one

Via an external form, I am creating/editing an item that I save at the server side.
Because the grid is a view that combines several entities, I cannot directly add a row, or even update an existing one.

This explains why I would really like to force a refresh (aka a fetch).

The only solution I could find was to update the PlutoLazyPaginationState and make it external, so that I can use a GlobalKey<PlutoLazyPaginationState> and invoke the key.currentState?.setPage(1);
But this is not ideal.

Many thanks

**Comment by RADeveloping** [comment]
Try using the setFilter method on the PlutoGridStateManager. 

`stateManager.setFilter((final element) => true);`

**Comment by doonfrs** [comment]
if key.currentState?.setPage(1); works why do you think it is not ideal?
